specs/sharding/beacon-chain.md

```diff
@@ -103,460 +101,377 @@ The following values are (non-configurable) constants used throughout the specif
-    shard_buffer: Vector[List[ShardWork, MAX_SHARDS], SHARD_STATE_MEMORY_SLOTS]
-    shard_sample_price: uint64
-```
+class IntermediateBlockBid(Container):
+    execution_payload_root: Root
 
-## New containers
+    sharded_data_commitment_root: Root # Root of the sharded data (only data, not beacon/intermediate block commitments)
 
-### `Builder`
+    sharded_data_commitment_count: uint64 # Count of sharded data commitments
 
-```python
-class Builder(Container):
-    pubkey: BLSPubkey
-    # TODO: fields for either an expiry mechanism (refunding execution account with remaining balance) 
-    #  and/or a builder-transaction mechanism.
+    bid: Gwei # Block builder bid paid to proposer
+    
+    # Block builders use an Eth1 address -- need signature as
+    # block builder fees and data gas base fees will be charged to this address
+	signature_y_parity: bool
+	signature_r: uint256
+	signature_s: uint256    
 ```
 
-### `DataCommitment`
+#### `ShardedCommitmentsContainer`
 
 ```python
-class DataCommitment(Container):
-    # KZG10 commitment to the data
-    point: BLSCommitment
-    # Length of the data in samples
-    samples_count: uint64
-```
+class ShardedCommitmentsContainer(Container):
+    sharded_commitments: List[KZGCommitment, 2 * MAX_SHARDS]
 
-### `AttestedDataCommitment`
+    # Aggregate degree proof for all sharded_commitments
+    degree_proof: KZGCommitment
 
-```python
-class AttestedDataCommitment(Container):
-    # KZG10 commitment to the data, and length
-    commitment: DataCommitment
-    # hash_tree_root of the ShardBlobHeader (stored so that attestations can be checked against it)
-    root: Root
-    # The proposer who included the shard-header
-    includer_index: ValidatorIndex
-```
+    # The sizes of the blocks encoded in the commitments (last intermediate and all beacon blocks since)
+    included_beacon_block_sizes: List[uint64, MAX_BEACON_BLOCKS_BETWEEN_INTERMEDIATE_BLOCKS + 1]
+    
+    # Number of commitments that are for sharded data (no blocks)
+    included_sharded_data_commitments: uint64
 
-### `ShardBlobBody`
+    # Random evaluation of beacon blocks + execution payload (this helps with quick verification)
+    block_verification_kzg_proof: KZGCommitment
+```
 
-Unsigned shard data, bundled by a shard-builder.
-Unique, signing different bodies as shard proposer for the same `(slot, shard)` is slashable.
+#### `IntermediateBlockBody`
 
 ```python
-class ShardBlobBody(Container):
-    # The actual data commitment
-    commitment: DataCommitment
-    # Proof that the degree < commitment.samples_count * POINTS_PER_SAMPLE
-    degree_proof: BLSCommitment
-    # The actual data. Should match the commitment and degree proof.
-    data: List[BLSPoint, POINTS_PER_SAMPLE * MAX_SAMPLES_PER_BLOB]
-    # fee payment fields (EIP 1559 like)
-    # TODO: express in MWei instead?
-    max_priority_fee_per_sample: Gwei
-    max_fee_per_sample: Gwei
+class IntermediateBlockBody(phase0.BeaconBlockBody):
```

Comment:
You are deriving from `phase0` block body and adding `execution_payload`. Does it mean you just want to exclude `sync_aggregate` from `IntermediateBlockBody`?

Reply:
I.e. why not the following class: 
```py
class IntermediateBlockBody(bellatrix.BeaconBlockBody):
    attestation: Attestation
    sharded_commitments_container: ShardedCommitmentsContainer
```

Reply:
This is not current anymore. Changed the structure so this discussion is obsolete.